[#47914] Rename Support Link and Dynamically set its URL

aaron-contreras opened the pull request:
https://community.openproject.org/work_packages/47914

- [x] Rename "Professional Support" links on the help menu and start page to "Enterprise Support"
- [x] While in a Community version, the help menu support link should point to the pricing page
- [x] While in an Enterprise edition, the help menu support link should point to the enterprise support docs.

**NOTE**: There was a caching bug that would keep a "Community version" help menu even
after an Enterprise Token was added and an upgrade to the Enterprise version was made. This
help menu would only be updated to an accurate state once the cache became invalidated.

Added a spec to cover both the new behavior and verify the bug is no longer present.

aaron-contreras committed:
Rename "Professional Support" links to "Enterprise Support"

The link to which a user is redirected when clicking on Enterprise Support
links is determined by whether or not they are on the Community or Enterprise Edition.

A user in Community Version gets redirected to the pricing page, while a user
on the Enterprise Edition gets redirected to the new landing page for Enterprise
Support docs.

aaron-contreras committed:
Account for Enterprise token in help menu's cache key

The cached help menu needs to be invalidated once an EnterpriseToken is set or
removed in order to properly render upsales/set appropriate support links.
Otherwise, an outdated up-sale link and Enterprise Support link would still be
displayed even when the OpenProject version toggles between Community and Enterprise.

oliverguenther commented on app/models/enterprise_token.rb:
```diff
@@ -41,6 +41,10 @@ def allows_to?(action)
+    end
+
     def show_banners?
       OpenProject::Configuration.ee_manager_visible? && (!current || current.expired?)
```
```suggestion
      OpenProject::Configuration.ee_manager_visible? && !active?
```

oliverguenther reviewed (approved):
Minor remark to reuse the newly added method below. Feel free to merge

aaron-contreras committed:
Re-use `.active?` in `show_banners?`

As the `active?` logic was extracted out, it makes sense to re-use this newly created method. Thanks Oliver!

Co-authored-by: Oliver Günther <o.guenther@openproject.com>
